src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -72,6 +79,204 @@ internal class Http2FrameWriter
         _scheduleInline = serviceContext.Scheduler == PipeScheduler.Inline;
 
         _hpackEncoder = new DynamicHPackEncoder(serviceContext.ServerOptions.AllowResponseHeaderCompression);
+
+        _channel = Channel.CreateBounded<Http2OutputProducer>(new BoundedChannelOptions(serviceContext.ServerOptions.Limits.Http2.MaxStreamsPerConnection)
+        {
+            AllowSynchronousContinuations = _scheduleInline
+        });
+
+        // This is null in tests sometimes
+        _window = connectionOutputFlowControl?.Available ?? 0;
+
+        _ = Task.Run(WriteToOutputPipe);
```

Comment:
Should this task be stored on a field and awaited when closing a connection?

Reply:
Yes. I need to find a good place to do this. There's a risk things that haven't shutdown properly don't run if this isn't running. I had it on Complete before but that was too early since Abort calls Complete.